### PR TITLE
[MPS] Add support for MPSProfiler Python bindings

### PR DESCRIPTION
**Pull request by razarmehr**
- Added torch.mps.profiler.[start() and stop()] APIs with RST documentation
- Added test case in test_mps

### CODE REVIEW

File: test/test_mps.py [hunk 1]
```diff
@@ -7301,6 +7301,26 @@ def test_mps_allocator_module(self):
...
+    # like this example:
+    # "aten::mps_convolution_backward_input:f32[1,128,6,6]:f32[128,64,3,3]:1,128,6,6 (id=G2, run=2)"
+    def test_mps_profiler_module(self):
+        torch.mps.profiler.start(mode="event", wait_until_completed=False)
+        # just running some ops to capture the OS Signposts traces for profiling
+        net1 = torch.nn.ConvTranspose2d(128, 64, kernel_size=3, stride=2, padding=1, output_padding=1)\
+            .to(device='mps', dtype=torch.float)
+        x = torch.rand(1, 128, 6, 6, device='mps', dtype=torch.float, requires_grad=True)
+        x = net1(x)
+        torch.mps.profiler.stop()
```
Comment (malfet) [inline test/test_mps.py:7316]: Let's have at least one test for context manager you are introducing
```suggestion
        with torch.mps.profiler.profile(mode="event", wait_until_completed=False) as p:
            # just running some ops to capture the OS Signposts traces for profiling
            net1 = torch.nn.ConvTranspose2d(128, 64, kernel_size=3, stride=2, padding=1, output_padding=1)\
                .to(device='mps', dtype=torch.float)
            x = torch.rand(1, 128, 6, 6, device='mps', dtype=torch.float, requires_grad=True)
            x = net1(x)
```

**Review by malfet — APPROVED** [review]
Looks ok, but please add at least one test for `profile` context manager

File: test/test_mps.py (hunk 1 above)
Comment (razarmehr) [reply test/test_mps.py:7316]: Thanks for the suggestion.

**Comment by razarmehr** [comment]
> Looks ok, but please add at least one test for `profile` context manager

Thanks for the review. I'll try to find a way for verification and will update the PR (if I find any).

**Comment by razarmehr** [comment]
For the verification test case, I'll send another PR to check if tracing is enabled using [os_signpost_enabled()](https://developer.apple.com/documentation/os/2976320-os_signpost_enabled) method.

**Pull request closed without merging**